lib/runCode.js

```diff
@@ -63,21 +65,22 @@ module.exports = function (opts, cb) {
     highestMemCost: new BN(0),
     depth: opts.depth || 0,
     suicides: opts.suicides || {},
-    block: opts.block || new Block(),
+    block: block,
     callValue: opts.value || new BN(0),
     address: opts.address || utils.zeros(32),
     caller: opts.caller || utils.zeros(32),
     origin: opts.origin || opts.caller || utils.zeros(32),
     callData: opts.data || new Buffer([0]),
     code: opts.code,
-    populateCache: opts.populateCache === undefined ? true : opts.populateCache
+    populateCache: opts.populateCache === undefined ? true : opts.populateCache,
+    enableHomestead: this.enableHomestead === undefined ? block.isHomestead() : this.enableHomestead // this == vm
```

Comment:
i think you can get away with  `this.enableHomestead || block.isHomestead()` here